File: Assets/Main/Scripts/Level/Level.cs
using UnityEngine;
using System.Collections;
using System;
using System.Collections.Generic;
using FullSerializer;

[Serializable]
public class Level
{
	public int base_x;
	public int base_y;
	public LevelLayer[] layers;
}

[Serializable]
public class LevelLayer
{
	public int order;
	public TextureToCell[] cells;
	public bool? walkable;
}

[Serializable]
public class TextureToCell
{
    public string texture;
    public CellInfo[] cells;
}

[Serializable]
public class CellInfo
{
	public int x_location;
	public int y_location;
	public bool? walkable;
}

File: Assets/API Server/level.go
package main

type Level struct {
	BaseX  uint16       `json:"base_x"`
	BaseY  uint16       `json:"base_y"`
	Layers []LevelLayer `json:"layers"`
}

type LevelLayer struct {
	Order uint8 `json:"order"`
	//Cells map[string][]CellInfo `json:"cells"`
	Cells    []TextureToCell `json:"cells"`
	Walkable *bool           `json:"walkable"`
}

type TextureToCell struct {
	Texture string     `json:"texture"`
	Cells   []CellInfo `json:"cells"`
}

type CellInfo struct {
	XLocation int8  `json:"x_location"`
	YLocation int8  `json:"y_location"`
	Walkable  *bool `json:"walkable"`
}
